### PR TITLE
Uml-985 frontend session check

### PR DESCRIPTION
**Pull request by Raffers**
# Purpose

Implements the front end part of session checking for users, no translation required as the displayed prompt has already been translated.

Fixes UML-985

## Approach

Adjusts the already implemented JS to make a call to the backend endpoint to confirm the current state of the session & checks if it at the warning threshold

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] The product team have tested these changes

### CODE REVIEW

**Review by GemTay — APPROVED** [review]

**Review by GemTay — CHANGES_REQUESTED** [review]
After manually testing, the popup displayed correctly, however when I clicked "Sign out" on the popup box, it did not sign me out

File: service-front/app/src/Common/templates/partials/session-dialog.html.twig [hunk 1]
```diff
@@ -11,7 +11,7 @@
     <button class="govuk-button govuk-!-margin-bottom-1 govuk-!-margin-right-1 jsHideTimeout" aria-label="Close Navigation">
         {% trans %}Stay signed in{% endtrans %}
     </button>
-    <a class="govuk-button govuk-button--secondary govuk-!-margin-bottom-1" href="{{ path('home') }}">
+    <a class="govuk-button govuk-button--secondary govuk-!-margin-bottom-1" href="{{ path('logout') }}">
```
Comment (GemTay) [inline service-front/app/src/Common/templates/partials/session-dialog.html.twig:14]: Fixed sign out issue here

**Review by GemTay — APPROVED** [review]

**Review by johnnolan — DISMISSED** [review]

File: service-front/web/src/index.js [hunk 2]
```diff
@@ -19,5 +19,7 @@ jsEnabled(document.body);
 disableButtonOnClick(document.getElementsByTagName('form'));
 new cookieConsent(document.getElementsByClassName('cookie-banner')[0], window.location.pathname === '/cookies');
 copyAccessCode();
-// new sessionDialog(document.getElementById("dialog"), 20); // TODO: Disabled for now until we know how we are going to trigger it
 showHidePassword();
+if (document.getElementsByClassName('js-signed-in').length > 0) {
```
Comment (johnnolan) [inline service-front/web/src/index.js:23]: Now its only for the actor side, you probably want to do an additional check for the dialog id as well so the JS doesn't run?

`document.getElementsByClassName('js-signed-in').length > 0 && document.getElementsById('dialog').length > 0`

File: service-front/app/src/Viewer/src/Handler/ViewerSessionCheckHandler.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $user = $this->getUser($request);
+        $session = $this->getSession($request, 'session');
+
+        $expiresAt = $session->get(EncryptedCookiePersistence::SESSION_TIME_KEY) + $this->sessionTime;
+        $timeRemaining = $expiresAt - time();
+        $sessionWarning = false;
+
+        if ($user !== null && $timeRemaining <= 300) {
```
Comment (cooperaj) [inline service-front/app/src/Viewer/src/Handler/ViewerSessionCheckHandler.php:59]: Same here

File: service-front/web/src/index.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ import copyAccessCode from './javascript/copyAccessCode';
 import cookieConsent from './javascript/cookieConsent';
 import sessionDialog from './javascript/sessionDialog';
 import showHidePassword from  './javascript/showHidePassword';
+import {doc} from "prettier";
```
Comment (johnnolan) [inline service-front/web/src/index.js:10]: Accidental import

**Review by johnnolan — CHANGES_REQUESTED** [review]

**Review by johnnolan — APPROVED** [review]

**Review by cooperaj — APPROVED** [review]
Solid

**Pull request merged**